### PR TITLE
Added Series.dt.is_month_end

### LINKED ISSUE #6396: [QST] is_month_end
I'm working on datetimes and have a simple is_month_start up and running but obviously is_month_end requires a bit more tinkering given different month lengths.

Approach I had was take a dataframe column and using .dt.day on it. Works fine but as soon as I try isin or a mask it doesn't like it. Series boolean compute issue.

Any help would be greatly appreciated.

### PR DESCRIPTION
**Pull request by TravisHester**
closes #6396

Adds support for Series.dt.is_month_end the same way it is used in pandas.

### CODE REVIEW

File: python/cudf/cudf/tests/test_datetime.py [hunk 1]
```diff
@@ -1326,3 +1326,32 @@ def test_is_month_start(data, dtype):
...
+            "2020-05-31",
+            None,
+            "1999-12-01",
+            "2000-12-21",
+            None,
+            "1900-02-28",
+            "1800-03-14",
+            "2100-03-10",
+            "1970-01-01",
+            "1969-12-11",
```
Comment (shwina) [inline python/cudf/cudf/tests/test_datetime.py:1344]: It might be a good idea to add `2020-09-29` here as a leap day test.

**Review by shwina — APPROVED** [review]
LGTM, one minor suggestion that shouldn't really be a blocker for merging.

File: python/cudf/cudf/tests/test_datetime.py (hunk 1 above)
Comment (TravisHester) [reply python/cudf/cudf/tests/test_datetime.py:1345]: Just added a leap day test.

File: python/cudf/cudf/core/series.py [hunk 2]
```diff
@@ -6006,6 +6006,24 @@ def days_in_month(self):
...
 
+    @property
+    def is_month_end(self):
+        """
+        Boolean indicator if the date is the last day of the month.
+
+        Returns
+        -------
+        Series
+        Booleans indicating if dates are the last day of the month.
```
[the remark this answers is not in the document]
Comment (TravisHester) [reply python/cudf/cudf/core/series.py:6017]: Just added an example.

**Review by galipremsagar — APPROVED** [review]

**Comment by galipremsagar** [comment]
@gpucibot merge

File: python/cudf/cudf/core/series.py [hunk 3]
```diff
@@ -6006,6 +6006,52 @@ def days_in_month(self):
...
+        1    False
+        2    False
+        3    False
+        4    False
+        5     True
+        6    False
+        7    False
+        8    False
+        dtype: bool
+        """
```
Comment (galipremsagar) [inline python/cudf/cudf/core/series.py:6046]:
```suggestion
        """  # noqa: E501
```

**Comment by galipremsagar** [comment]
@gpucibot merge

**Pull request merged**